fitstack/signal.py

```diff
@@ -648,13 +649,25 @@ def load_from_ps2Dfiles(
 
         # Find directories which match the right format
         for d in sorted(dirs):
-            mo = re.search(r"_compderiv-([^\/]+)", d)
-
-            if mo is None:
+            ###mo = re.search(r"_compderiv-([^\/]+)", d)
+            mo_comp = re.search(r"template_bias_([0-9\.]+)_Pk_([^_]+)_ps_HI", d)
+            # Check for shotnoise directory
+            mo_shotnoise = re.search(r"template_shotnoise", d)
+
+            if mo_comp:
+                bias = mo_comp.group(1)  # This will be "0", "0.5", or "1"
+                pk_type = mo_comp.group(2)  # This will be "base", "lin", "FoGh"
+                # Create a composite key that identifies the templates
+                key = f"{bias}-{pk_type}"
+
+            elif mo_shotnoise:
+                key = "shotnoise"
+            else:
                 print(f"Directory {d} does not match expected format, rejecting")
                 continue
 
-            key = mo.group(1)
+            print(f"Processing directory: {d}")
```

Comment:
```suggestion
            logger.debug(f"Processing directory: {d}")
```
Let's move to using `logger` instead of `print`. (There's a mix of `logger` and `print` throughout `fitstack`, and we should eventually change all the other `print` statements to use `logger` for consistency.)